src/main/java/com/rapid7/client/dcerpc/mssamr/SecurityAccountManagerService.java

```diff
@@ -760,6 +764,24 @@ public SID getSIDForDomain(final ServerHandle serverHandle, final String domainN
         return response.getList();
     }
 
+    public MembershipWithNameUse[] getIDsForDomain(final DomainHandle domainHandle, int... rids) throws IOException {
+        final SamrLookupIdsInDomainRequest request = new SamrLookupIdsInDomainRequest(parseHandle(domainHandle), rids);
+        final SamrLookupIdsInDomainResponse response = callExpect(request, "SamrLookupIdsInDomain",
+            SystemErrorCode.ERROR_SUCCESS, SystemErrorCode.STATUS_SOME_NOT_MAPPED, SystemErrorCode.STATUS_NONE_MAPPED);
+        if (SystemErrorCode.STATUS_NONE_MAPPED.getValue() == response.getReturnValue())
+            return new MembershipWithNameUse[0];
+
+        List<NonNullTerminated> names = response.getNames();
+        long[] array = response.getUses().getArray();
+        List<MembershipWithNameUse> members = new LinkedList<>();
```

Comment:
There's no need to use a list here, the size is known upfront; use `MembershipWithNameUse[]`

Reply:
There's null entry that we don't return based on what we did in lookupnames.

Reply:
But we want to maintain the null entries do we not? Otherwise you won't know what mapped. i.e. I believe this is wrong:
```
if (names.get(i) != null) {
```

Reply:
If the name is null we (or the rid is null, if it can be), it should be null entry